File: java/algorithm/src/leetcode/RegularExpressionMatching_10.java
package leetcode;

// #10 - Regular Expression Matching
import java.util.regex.Pattern;

public class RegularExpressionMatching_10 {
	public static void main(String[] args) {
		String s = "aa";
		String p = ".*";

		System.out.println("isMatch(s, p) = " + isMatch(s, p));
	}

	public static boolean isMatch(String s, String p) {
		return Pattern.compile(p).matcher(s).matches();
	}
}

File: ruby/problem_solving/leetcode/RegularExpressionMatching_10.rb
# ruby 10 Regular Expression Matching

# @param {String} s
# @param {String} p
# @return {Boolean}
def is_match(s, p)
  s =~ /(#{p})/
  s == $1
end

s = "ab"
p = ".*"
p is_match(s, p)